index.d.ts

```diff
@@ -0,0 +1,91 @@
+import { ITypedField, ITypedValue, ITypedData } from 'eth-sig-util';
+import { Transaction } from 'ethereumjs-tx';
+export { ITypedField, ITypedValue, ITypedData }
+
+type WalletOpts = {
+  withAppKeyOrigin?: string
+}
+
+export interface IKeyring {
+  serialize(): Promise<any>
+  deserialize(serialized:any): Promise<void>
+
+  addAccounts(numberToAdd: number): Array<string> // returns address array
+  getAccounts(): Promise<Array<string>>
+  removeAccount?(address:string): Promise<void>
+
+  // Implementing getPrivateKeyFor allows inheriting eth-simple-keyring for free signing method support.
+  getPrivateKeyFor?(address:string, opts?: WalletOpts): Buffer
+
+  // If private key method above is not provided, these methods will need to be implemented per-implementation.
+  signTransaction?(address:string, tx: Transaction, opts?: WalletOpts): Promise<string>
```

Comment:
Shouldn't these still be mandatory? If they're obtained via inheritance, they'd still be on the class.